sample/src/samples/datepicker/validation.html

```diff
@@ -0,0 +1,34 @@
+<template>
+  <md-card md-title="Basic">
+    <div>
+      <input md-datepicker="container: body; value.two-way: date & validate:rules;" type="date" placeholder="pick a date">
+    </div>
+    <div style="margin-top: 15px;">
+      <button md-waves md-button click.delegate="validateModel()">validate</button>
+      <button md-waves md-button click.delegate="reset()">reset</button>
+    </div>
+  </md-card>
+
+  <md-card md-title="Show or hide error text">
+    <div>
+      <input md-datepicker="show-errortext.bind: showErrortext; container: body; value.two-way: date2 & validate:rules;" type="date" placeholder="pick a date">
+    </div>
+    <div style="margin-top: 15px;">
+      <button md-waves md-button click.delegate="validateModel2()">validate</button>
+      <button md-waves md-button click.delegate="reset2()">reset</button>
+    </div>
+    <md-switch style="margin-top: 15px;" md-label-on="show" md-label-off="hide" md-checked.bind="showErrortext"></md-switch>
+  </md-card>
+
+  <md-card if.bind="controller.errors.length">
+    <h5 class="error-text">You have errors!</h5>
+    <ul style="margin-top: 15px;">
+      <li repeat.for="error of controller.errors">
+        <a href="#" click.delegate="error.target.focus()">
```

Comment:
The pickadate api docs suggest that the input is focused when opening:
http://amsul.ca/pickadate.js/api/#method-open-focus

I guess the input is focused, then the picker is opened, causing the input to `blur`.
It seems we could work around that by opening/closing the picker explicitly and override the built-in mechanic (but I may be wrong as I've only read briefly).

"Rewriting" the open/close mechanic to be explicit seems to be a lot (consider testing all that)..
I think having an error message when the picker opens doesn't stop us from adding these changes. At least for now, that would be okay.

What do you think?